app/models/conference.rb

```diff
@@ -317,6 +317,57 @@ def event_distribution
   end
 
   ##
+  # Returns a hash with scheduled vs unscheduled events
+  # { "Scheduled" => { value: number of confirmed and scheduled events, color: color },
+  #   "Unscheduled" => { value: number of confirmed and unscheduled events, color: color }
+  #
+  # ====Returns
+  # * +hash+ -> hash
+  def scheduled_event_distribution
+    confirmed_events = program.events.where(state: 'confirmed')
+    scheduled_value =  { 'value' => confirmed_events.where.not(start_time: nil).count, 'color' => 'green' }
+    unscheduled_value =  { 'value' => confirmed_events.where(start_time: nil).count, 'color' => 'red' }
+    { 'Scheduled' => scheduled_value, 'Unscheduled' => unscheduled_value }
+  end
+
+  ##
+  # Returns a hash with Registration attended vs. Registration not attended
+  # { "Attended" => { value: number of registration attended, color: color },
+  #   "Not attended" => { value: number of registration not attended, color: color }
+  #
+  # ====Returns
+  # * +hash+ -> hash
+  def registration_distribution
+    reg = registrations.includes(:user)
+    attended_value =  { 'value' => reg.where(attended: true).count, 'color' => 'magenta' }
+    not_attended_value =  { 'value' => reg.where.not(attended: true).count, 'color' => 'blue' }
+    { 'Attended' => attended_value, 'Not attended' => not_attended_value }
+  end
+
+  ##
+  # Returns a hash with affiliation =>
+  # {value: count of registration whose user has that affilation, color: color}
+  # In case that the affiliation is nil, it groups it in not_attended_value
+  #
+  # ====Returns
+  # * +hash+ -> hash
+  def affiliation_distribution
+    counted_affiliations = registrations.joins(:user).group(:affiliation).count
+    result = {}
+    i=1
+    counted_affiliations.each do |key, value|
+      capitalized_key = (key.nil? ? 'None' : key.capitalize)
```

Comment:
The problem is that empty string  is grouped separated from nil. I can group them together